src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -315,6 +315,57 @@ public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
     return success;
   }
 
+  @Override
+  public CourseConfirmState getConfirmeState(DocumentReference regDocRef) {
+    DocumentReference courseParticipantClassRef = getCourseClasses().getCourseParticipantClassRef(
+        modelContext.getXWikiContext().getDatabase());
+    CourseConfirmState confirmState = CourseConfirmState.UNCONFIRMED;
+    try {
+      List<BaseObject> partiObjs = modelAccess.getXObjects(regDocRef, courseParticipantClassRef);
+      int index = 0;
+      for (BaseObject obj : partiObjs) {
+        Optional<CourseConfirmState> state = CourseConfirmState.convertStringToEnum(
+            obj.getStringValue("status"));
+        if (state.isPresent()) {
+          if (isConfirmed(state.get(), confirmState, index)) {
```

Comment:
I suggest to use a switch statement. Initialize confirmState with null. use following transition rules:
null -> state.get()
UNCONFIRMED, CONFIRMED if (state.get() != confirmState) -> PARTIALCONFIRMED
default -> nothing changes
if (confirmState == null) return UNCONFIRMED else return confirmState